storage/mongodb/webhook.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mongodb
+
+import (
+	"strings"
+
+	mgo "github.com/globalsign/mgo"
+	"github.com/globalsign/mgo/bson"
+	"github.com/tsuru/tsuru/db"
+	dbStorage "github.com/tsuru/tsuru/db/storage"
+	"github.com/tsuru/tsuru/types/event"
+)
+
+type webhookStorage struct{}
+
+func webhoookCollection(conn *db.Storage) *dbStorage.Collection {
+	coll := conn.Collection("webhook")
+	coll.EnsureIndex(mgo.Index{
+		Key:    []string{"name"},
+		Unique: true,
+	})
+	return coll
+}
+
+var _ event.WebHookStorage = &webhookStorage{}
+
+func (s *webhookStorage) Insert(w event.WebHook) error {
+	conn, err := db.Conn()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	err = webhoookCollection(conn).Insert(w)
+	if err != nil && mgo.IsDup(err) {
+		err = event.ErrWebHookAlreadyExists
+	}
+	return err
+}
+
+func (s *webhookStorage) Update(w event.WebHook) error {
+	conn, err := db.Conn()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+	err = webhoookCollection(conn).Update(bson.M{"name": w.Name}, w)
+	if err == mgo.ErrNotFound {
+		err = event.ErrWebHookNotFound
+	}
+	return err
+}
+
+func (s *webhookStorage) findQuery(query bson.M) ([]event.WebHook, error) {
+	conn, err := db.Conn()
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+	var webhooks []event.WebHook
+	err = webhoookCollection(conn).Find(query).All(&webhooks)
+	return webhooks, err
+}
+
+func (s *webhookStorage) FindAllByTeams(teams []string) ([]event.WebHook, error) {
+	var query bson.M
+	if teams != nil {
+		query = bson.M{"teamowner": bson.M{"$in": teams}}
+	}
+	return s.findQuery(query)
+}
+
+func (s *webhookStorage) FindByEvent(f event.WebHookEventFilter, isSuccess bool) ([]event.WebHook, error) {
+	for _, name := range f.KindNames {
+		parts := strings.Split(name, ".")
+		if len(parts) == 0 {
```

Comment:
Is there a case where `len(parts)` is 0?

Reply:
Nope, removing this